content/en/blog/2024/elastic-contributes-continuous-profiling-agent.md

```diff
@@ -0,0 +1,149 @@
+- Disruptive service restarts
+- Inability to get visibility into third-party libraries
+
+Continuous profiling, however, runs in the background with minimal overhead,
+providing real-time, actionable insights without the need to replicate issues in
+separate environments.
+
+This allows SREs, DevOps, and developers to see how code affects performance and
+cost, making code and infrastructure improvements easier.
+
+## Contribution of comprehensive profiling abilities
+
+The continuous profiling agent, that Elastic is donating, is
+[based on eBPF](https://ebpf.io/) and by that a whole system, always-on solution
+that observes code and third-party libraries, kernel operations, and other code
+you don't own. It eliminates the need for code instrumentation
+(run-time/bytecode), recompilation, or service restarts with low overhead, low
+CPU (~1%), and memory usage in production environments.
+
+The donated profiling agent facilitates identifying non-optimal code paths,
+uncovering "unknown unknowns", and provides comprehensive visibility into the
+runtime behavior of all applications. The continuous profiling agent provides
+support for a wide range of runtimes and languages, such as:
+
+- C/C++
+- Rust
+- Zig
+- Go
+- Java
+- Python
+- Ruby
+- PHP
+- Node.js / V8
+- Perl
+- .NET
+
+Additionally, organizations can combine profiling data with resource information
+to derive an estimate for the carbon footprints of services. This helps
+organizations meet their sustainability objectives by minimizing computational
+wastage, ensuring seamless alignment with their strategic ESG goals.
+
+## Benefits to OpenTelemetry
+
+This contribution not only boosts the standardization of continuous profiling
+for observability but also accelerates its adoption as a key signal in
+OpenTelemetry. Customers benefit from a vendor-agnostic method of collecting
+profiling data correlating it with existing signals, like tracing, metrics, and
+logs, opening new potential for observability insights and a more efficient
+troubleshooting experience.
+
+### User Benefits of OpenTelemetry Profiling
+
+OpenTelemetry-based continuous profiling unlocks the following possibilities for
+users:
+
+- Continuous profiling data compliments the existing signals (traces, metrics
+  and logs) by providing detailed, code-level insights on the services'
+  behavior.
+
+- Seamless correlation with other OpenTelemetry signals such as traces,
+  increasing fidelity and investigatory depth.
+
+- Minimize environmental impact: energy consumption associated with computing is
+  a growing concern (source:
+  [MIT Energy Initiative](https://energy.mit.edu/news/energy-efficient-computing/)).
+  More efficient code translates to lower energy consumption, contributing to a
+  reduction in carbon (CO2) footprint. Combining profiling data with
+  OpenTelemetry's resource information (i.e. resource attributes) allows to
+  derive insights into the services' carbon footprint.
```

Comment:
```suggestion
- Estimate environmental impact: Combining profiling data with
  OpenTelemetry's resource information (i.e. resource attributes) allows to
  derive insights into the services' carbon footprint.
```

Observability provides the data that allows to estimate, analyze, etc, but it does not minimize the environmental impact directly. "Enable minimization of environmental impact" would also be possible.